Cesium.Runtime/StringFunctions.cs

```diff
@@ -192,4 +193,19 @@ public static int MemCmp(void* lhs, void* rhs, nuint count)
 
         return 0;
     }
+
+    public static UTF8String StrDup(UTF8String src)
+    {
+        return StrNDup(src, src.Length + 1);
+    }
+
+    public static UTF8String StrNDup(UTF8String src, nuint count)
+    {
+        if (src.Pointer == null)
+            return UTF8String.NullString;
+
+        var dest = new UTF8String((byte*)StdLibFunctions.Malloc(count));
+        src.CopyTo(dest, count);
+        return dest;
```

Comment:
I think this misses the null terminator. Please note this wording from the standard:
> If the array pointed to by `s` does not contain a null within the first size
characters, a null is appended to the copy of the array.

Meaning the result should always be null-terminated, regardless of `src[count]` being a null byte or not.